src/Illuminate/Cache/DatabaseStore.php

```diff
@@ -149,23 +145,32 @@ public function decrement($key, $value = 1)
      * @param  string  $key
      * @param  mixed  $value
      * @param  \Closure  $callback
-     * @return void
+     * @return int|bool
      */
     protected function incrementOrDecrement($key, $value, Closure $callback)
     {
-        $prefixed = $this->prefix.$key;
+        return $this->connection->transaction(function () use ($key, $value, $callback) {
+            $prefixed = $this->prefix.$key;
 
-        $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
+            $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
+
```

Comment:
I do wonder if sharedLock could lead to race conditions now that I think
about it though.

On Wed, Dec 30, 2015 at 9:07 AM, Will Rowe <notifications@github.com> wrote:

> In src/Illuminate/Cache/DatabaseStore.php
> <https://github.com/laravel/framework/pull/11582#discussion_r48608380>:
>
> >
> > -        $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
> > +            $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
> > +
>
> Using sharedLock() should be fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laravel/framework/pull/11582/files#r48608380>.
>